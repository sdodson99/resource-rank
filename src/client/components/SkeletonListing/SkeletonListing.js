import React from 'react';
import Skeleton from 'react-loading-skeleton';
import styles from './SkeletonListing.module.css';

const SkeletonListing = () => (
  <div className={styles.SkeletonListing} data-testid="SkeletonListing">
    <Skeleton className="mb-3" count={3} height={100} />
  </div>
);

export default SkeletonListing;
