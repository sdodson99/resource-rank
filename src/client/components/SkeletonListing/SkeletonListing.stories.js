import React from 'react';
import SkeletonListing from './SkeletonListing';

export default {
  component: SkeletonListing,
  title: 'Components/SkeletonListing',
};

const Template = (args) => <SkeletonListing {...args} />;

export const Default = Template.bind({});
Default.args = {};
