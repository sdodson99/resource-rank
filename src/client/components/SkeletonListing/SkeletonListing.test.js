import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { createRenderer } from 'react-test-renderer/shallow';
import SkeletonListing from './SkeletonListing';

describe('<SkeletonListing />', () => {
  it('should mount', () => {
    render(<SkeletonListing />);

    const skeletonListing = screen.getByTestId('SkeletonListing');

    expect(skeletonListing).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const component = createRenderer().render(<SkeletonListing />);

    expect(component).toMatchSnapshot();
  });
});
