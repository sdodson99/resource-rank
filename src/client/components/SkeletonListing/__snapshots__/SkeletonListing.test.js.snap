// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkeletonListing /> should render correctly 1`] = `
<div
  className="SkeletonListing"
  data-testid="SkeletonListing"
>
  <Skeleton
    circle={false}
    className="mb-3"
    count={3}
    delay={0}
    duration={1.2}
    height={100}
    width={null}
    wrapper={null}
  />
</div>
`;
